validator/validator.go

```diff
@@ -0,0 +1,73 @@
+package validator
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"gopkg.in/Clever/kayvee-go.v2/logger"
+)
+
+var (
+	requiredKayveeFields = []string{
+		"source",
+		"title",
+		"level",
+	}
+
+	validLevels = map[string]bool{
+		logger.Debug.String():    true,
+		logger.Info.String():     true,
+		logger.Warning.String():  true,
+		logger.Error.String():    true,
+		logger.Critical.String(): true,
+	}
+)
+
+// ValidateJSONFormat returns an error if the given string is not a valid
+// JSON-formatted kayvee log line.
+func ValidateJSONFormat(logLine string) error {
+	var kayveeData map[string]interface{}
+
+	err := json.Unmarshal([]byte(strings.TrimSpace(logLine)), &kayveeData)
+	if err != nil {
+		return fmt.Errorf("Invalid JSON: %s", err)
+	}
+
+	return validateKayveeData(kayveeData)
+}
+
+func validateKayveeData(kayveeData map[string]interface{}) error {
+	requiredFields := requiredKayveeFields
+
+	// Add type-specific required fields, if applicable.
+	if kayveeData["type"] != nil {
+		logType, isString := kayveeData["type"].(string)
+		if !isString {
+			return fmt.Errorf("Invalid log type `%s`", kayveeData["type"])
```

Comment:
More simply, could log in the testcase what field is missing / why it should be failing